front/app/modules/commercial/content_builder/admin/containers/index.tsx

```diff
@@ -21,34 +24,104 @@ import ContentBuilderSettings from '../components/ContentBuilderSettings';
 import { PROJECT_DESCRIPTION_CODE } from '../../services/contentBuilder';
 import useLocale from 'hooks/useLocale';
 import useContentBuilderLayout from '../../hooks/useContentBuilder';
+import useFeatureFlag from 'hooks/useFeatureFlag';
+import useAppConfigurationLocales from 'hooks/useAppConfigurationLocales';
 
 // utils
 import { isNilOrError } from 'utils/helperUtils';
 import { SerializedNodes } from '@craftjs/core';
+import { Locale } from 'typings';
+import eventEmitter from 'utils/eventEmitter';
+
+// intl
+import messages from '../messages';
+import FormattedMessage from 'utils/cl-intl/FormattedMessage';
 
 const StyledRightColumn = styled(RightColumn)`
-  min-height: calc(100vh - ${2 * stylingConsts.menuHeight}px);
+  min-height: calc(100vh - ${stylingConsts.menuHeight}px);
   z-index: 2;
   margin: 0;
   max-width: 100%;
   align-items: center;
   padding-bottom: 100px;
 `;
 
-const ContentBuilderPage = ({ params: { projectId } }: WithRouterProps) => {
+type ContentBuilderErrors = Record<
+  string,
+  { hasError: boolean; selectedLocale: Locale }
+>;
+
+export const ContentBuilderPage = () => {
   const [mobilePreviewEnabled, setMobilePreviewEnabled] = useState(false);
+  const [selectedLocale, setSelectedLocale] = useState<Locale | undefined>();
+  const [draftData, setDraftData] = useState<Record<string, SerializedNodes>>();
+  const { pathname } = useLocation();
+  const { projectId } = useParams() as { projectId: string };
+
   const iframeRef = useRef<HTMLIFrameElement | null>(null);
 
+  const featureEnabled = useFeatureFlag({ name: 'content_builder' });
   const locale = useLocale();
-
+  const locales = useAppConfigurationLocales();
   const contentBuilderLayout = useContentBuilderLayout({
     projectId,
     code: PROJECT_DESCRIPTION_CODE,
   });
 
+  useEffect(() => {
+    if (!isNilOrError(locale)) {
+      setSelectedLocale(locale);
+    }
+  }, [locale]);
+
+  const [contentBuilderErrors, setContentBuilderErrors] =
+    useState<ContentBuilderErrors>({});
+
+  useEffect(() => {
+    const subscription = eventEmitter
+      .observeEvent('contentBuilderError')
```

Comment:
Minor: What do you think about using constants for the event names 🤔 ?

Reply:
Very good point - added!